### DISCUSSION TITLE
How Can I Loop Through All Entities or Community Summaries in GraphRAG ?

### DISCUSSION THREAD

**Discussion by octadion** [microsoft/graphrag#1304] [Q&A] [answered]
Hi everyone,

In graphrag, can I loop through all entities or community summaries for.. lets say to generate synthetic data, using the entire entity or community summary as a context.

Could anyone guide me on how to accomplish this?

thankyou

**Comment by Adriansdls** [comment, marked as answer]
Hi! Yes, you can do that. The parquet files that you get after indexing can be opened in a pandas dataframe. Once in the pandas dataframe you can very easily loop through all entities. Check the local search notebook in the repo, you will find what you need to get started.
